msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,136 @@
+
+- Tabs: personal (`staticTabs`) and configurable (`configurableTabs`)
+- Message extensions (`composeExtensions`), including individual commands
+- Bots (`bots`)
+
+Each app element is denoted by an `id`, which maps to `botId` for bots, `entityId` for static tabs, and `id` for configurable tabs and message extensions.
+
+This feature introduces `id` properties to `configurableTabs` and `composeExtensions` definitions. Teams and other Microsoft 365 hosts support apps that contain either a single configurable tab or a single message extension, or both. These ID properties future-proof your app in case host support expands to accommodate multiple instances of these components.
+> [!NOTE]
+> The `id` property must be specified in a `configurableTab` or a `composeExtension` for it to be recognized in an `elementRelationshipSet`.
+
+### One-way dependencies
+
+Use the [`oneWayDependencies`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipsetonewaydependency) array to describe cases where one component of your app depends upon another component. For each object in the array, specify the dependent component (`element`) and the component it depends on (`dependsOn`). The following JSON snippet shows specific message extension commands that have a one-way dependency on a bot:
+
+```json
+    "elementRelationshipSet": {
+      "oneWayDependencies" : [
+        {
+          "element" : {
+            "name" : "composeExtensions",
+            "id" : "composeExtension-id",
+            "commandIds": ["command-1-id", "command-2-id"]
+          },
+          "dependsOn" : [
+              {"name" : "bots", "id" : "bot-id"}
+            ]
+        }
+      ]
+    }
+```
+
+For message extensions, you can optionally specify individual commands that require support for specific app components. If those components aren't supported in the runtime host, they won't be made available to the user (though all other commands will run).
+
+### Mutual dependencies
+
+Use the [`mutualDependencies`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipsetmutualdependencies) array to group app components that must load together in order to support their intended function. Each object in the array represents a mutually dependent app component. The following JSON snippet shows a bot, static tab, message extension, and configurable tab that are mutually dependent on each other:
+
+```json
+    "elementRelationshipSet": {
+      "mutualDependencies" : [
+                {"name" : "bots", "id" : "bot-id"}, 
+                {"name" : "staticTabs", "id" : "staticTab-id"},
+                {"name" : "composeExtensions", "id" : "composeExtension-id"},
+                {"name" : "configurableTabs", "id": "configurableTab-id"}
+      ]
+    },
+```
+
+## Specify runtime capability requirements for specific app components (`requirementSet`)
```

Comment:
```suggestion
## `requirementSet`
```

Reply:
Same here, especially because this article is about teaching customers about RT. Let's lead with customer scenario, and then introduce our tech